docs/building_blocks_for_flexible_content.md

```diff
@@ -0,0 +1,571 @@
+
+Blocks are added to the content item in the details hash. The landing page editor is currently a work in progress. At time of writing, the config for blocks need to be added in YAML format in Whitehall. This YAML config is then merged into `details` hash when the landing page is saved and published.
+
+The YAML config must include a top-level `blocks` element. e.g.
+
+```yaml
+blocks:
+- type: block_type
+```
+
+...and may include a top-level `navigation_groups` element for use by blocks (See [Navigation Groups](#navigation-groups))
+
+Block configuration is designed to be as flexible as possible, so blocks can be nested inside other blocks. An example configuration for each type of block is shown below.
+
+## Types of Block
+
+Each block rendered using its own partial template. The full list of available types of block available can be found by looking at the contents of: `app/views/landing_page/blocks`
+
+### Simple blocks
+
+Simple blocks general render one component or "thing". They can either be rendered directly or as the children of compound blocks.
+
+- [Action link](#action-link)
+- [Big number](#big-number)
+- [Document list](#document-list)
+- [Govspeak](#govspeak)
+- [Heading](#heading)
+- [Image](#image)
+- [Main Navigation](#main-navigation)
+- [Quote](#quote)
+- [Share links](#share-links)
+- [Side Navigation](#side-navigation)
+- [Statistics](#statistics)
+
+#### Action link
+
+A wrapper around the [action link component](https://components.publishing.service.gov.uk/component-guide/action_link).
+
+##### Example
+
+```yaml
+- type: action_link
+  text: "Learn more about our goals"
+  href: "/landing-page/goals"
+```
+
+#### Big number
+
+A wrapper around the [Big number component](https://components.publishing.service.gov.uk/component-guide/big_number)
+
+##### Example
+
+```yaml
+- type: big_number
+  number: £75m
+  label: amount of money that looks big
+```
+
+#### Document list
+
+A wrapper around the [Document list component](https://components.publishing.service.gov.uk/component-guide/document_list)
+
+##### Example
+
+```yaml
+- type: document_list
+  items:
+  - text: An example link
+    path: https://www.gov.uk
+    document_type: Press release
+    public_updated_at: "2016-06-27 10:29:44 +0000"
+  - text: Another example link
+    path: https://www.gov.uk
+    document_type: News article
+    public_updated_at: "2021-01-16 11:34:12 +0000"
+  - text: A third example link
+    path: https://www.gov.uk
+    document_type: Consultation
+    public_updated_at: "2024-02-01 09:00:11 +0000"
+```
+
+#### Govspeak
+
+A wrapper around the [Govspeak content component](https://components.publishing.service.gov.uk/component-guide/govspeak). Note that the content item data for the component is not Govspeak, but HTML that has been converted from Govspeak. If you're editing the block data in Whitehall, though, you can use Govspeak directly by taking advantage of the publishing-api automatic conversion (see second example)
+
```

Comment:
This reads as a bit confusing. This makes it sound like I need to go and find a tool to produce some HTML by converting it whereas in fact I can just supply any supported HTML. I don't think there's any need to mention publishing-api, just that there are two options for input: Any HTML which only uses the govspeak approved subset of tags, or govspeak. Then call out the syntax difference needed for each option as illustrated.

Reply:
Does this read better?

---

A wrapper around the [Govspeak content component](https://components.publishing.service.gov.uk/component-guide/govspeak). 

Content for this block can either be written as govspeak or HTML. If you are adding a Govspeak block in config in the `frontend` application, it must be added as HTML. This is because, despite the name, the Govspeak component it calls actually renders HTML content.

If you are adding the block in config in Whitehall you have a choice of writing it in either govspeak or HTML. That is because govspeak content added in Whitehall is automatically converted to HTML by publishing-api. However, if you use govspeak, you'll need to supply the `content-type` (see second example).

Reply:
Yes, much better.

I know that govspeak doesn't support all HTML and will drop some tags, would that affect stuff input here? I'm not sure what tags are unsupported. If it would affect though it's worth calling out.

Reply:
I didn't know that! The best I can do is link to the govspeak docs.

Reply:
I think it's fine as is.